webboot/webboot_test.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"encoding/hex"
+	"crypto/md5"
+	"io"
+	"io/ioutil"
+	"log"
+	"testing"
+	"reflect"
+
+)
+
+func TestWget(t *testing.T) {
+	type test struct {
+		name	string
+		isolink	string
+		md5link	string
+	}
+	tests := []test{
+		{name: "tinyCoreISO", isolink: "http://tinycorelinux.net/10.x/x86/release/Core-10.1.iso", md5link: "http://tinycorelinux.net/10.x/x86/release/Core-10.1.iso.md5.txt"},
+		{name: "ubuntuISO", isolink: "http://releases.ubuntu.com/bionic/ubuntu-18.04.2-desktop-amd64.iso", md5link: "http://releases.ubuntu.com/bionic/MD5SUMS"},
+		{name: "archLinuxISO", isolink: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/archlinux-2019.08.01-x86_64.iso", md5link: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/md5sums.txt"},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			reader, err := wget(v.isolink)
+			if err != nil {
+				t.Errorf("Failed to receive the ISO: got %v, want nil", err)
+			}
+			body, err := ioutil.ReadAll(reader)
+			hexa := hex.EncodeToString(body)
```

Comment:
This step really confuses me. Why are you encoding the body to a string?you can just md5.Sum the data.

Reply:
md5.Sum returns a byte array, and the hash I receive from md5.txt is a string. To compare the two among each other, I either to encode one or decode the other.

Reply:
func Sum(data []byte) [Size]byte
it takes a byte array. So why are you converting the body, which is a byte array, to a string? you want to md5sum the body, right?

Reply:
Ahh, I forgot that this is my old PR from Friday. I have made changes to this test. The thing is I was running into issues at this exact step because I was converting the body into a string before I inserted it into a md5.Sum() and it printed an error because strings cannot be in md5.Sum. This is why I pushed up my PR as a WIP. But shortly after, I realized why the error was occurring and changed my test to insert the body into md5.Sum AND THEN convert it into a string :) Sorry about the confusion! :)